src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -199,9 +201,12 @@ impl<'a> JobQueue<'a> {
     // In general, we try to print "Compiling" for the first nontrivial task
     // run for a package, regardless of when that is. We then don't print
     // out any more information for a package after we've printed it once.
-    fn note_working_on(&mut self, config: &Config, pkg: &'a PackageId,
+    fn note_working_on(&mut self, config: &Config, key: &Key<'a>,
                        fresh: Freshness) -> CargoResult<()> {
-        if self.printed.contains(&pkg) {
+        let pkg: &'a PackageId = key.pkg;
+        let doc: &bool = &key.profile.doc;
```

Comment:
It should be ok to not bind these to local variables, you can probably just use them through `key` directly.